portal-impl/src/com/liferay/portal/security/permission/AdvancedPermissionChecker.java

```diff
@@ -396,14 +397,16 @@ public boolean isOrganizationOwner(long organizationId) {
 	protected void addTeamRoles(long userId, Group group, Set<Long> roleIds)
 		throws Exception {
 
-		int count = TeamLocalServiceUtil.getGroupTeamsCount(group.getGroupId());
+		long groupId = GroupThreadLocal.getGroupId();
```

Comment:
Can we just do group.isStagingGroup() here and do the count/fetch with the live groupId?

Reply:
Hey @Preston-Crary,

I am not sure if I understand your question here but the Staging groupId for Teams since they are the only permission entities that have they handle the memberships independently in Staging and Live.

Please, provide the code example in case you still think that's needed.

Thank you.

@SamZiemer

Reply:
Umm I just realized that the code you pointed out is outdated, we don't use the groupThreadLocal anymore after @csierra changes

Reply:
Does this work for what you need? https://github.com/achaparro/liferay-portal/pull/624

Reply:
Hey @Preston-Crary,

I am afraid that that solution won't be valid since we need the current groupId, not the Staging group. So, if the user is in the staging site, use the staging groupId, if he is in the live group, use the live groupId. The latest point where we know which is the current groupId is here:
https://github.com/achaparro/liferay-portal/pull/623/commits/7eb7cfd1dde47400b559f7c83102743a8d8b20b5#diff-3669a38b746fb70cfdbe70441f0c1aeeR139


cc/ @samziemer @csierra